### PR TITLE
GOV302: Pre-processing validation of data (well formed data and completeness checks) prior to disbursement

### PR DESCRIPTION
**Pull request by ankita10r**
Implementation for [302](https://mifosforge.jira.com/browse/GOV-302)

Related PR:

https://github.com/openMF/ph-ee-env-labs/pull/106
https://github.com/openMF/ph-ee-connector-integration-test/pull/43

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [x] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [x] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java [hunk 1]
```diff
@@ -178,8 +178,8 @@ public void configure() {
                     extraVariables.put(NOTE,gsmaChannelRequest.getDescriptionText());
                     logger.info("Payee Tenant ID {}", extraVariables.get("payeeTenantId"));
                     String bpmn = getWorkflowForTenant(tenantId);
-                    String tenantSpecificBpmn = bpmn.equals("default")?baseTransaction:bpmn
-                            .replace("{dfspid}", tenantId);
+                    String tenantSpecificBpmn = bpmn.equals("default")?baseTransaction .replace("{dfspid}",
+                            tenantId):bpmn.replace("{dfspid}", tenantId);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java:182]: we had this argument before that the tenant is not same as dfspid and dfspid will come as a separate header: accountholdingInstitutionId

**Review by fynmanoj — CHANGES_REQUESTED** [review]
check the dfspid vs tenant logic

File: src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java (hunk 1 above)
Comment (ankita10r) [reply src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java:182]: This dfspid is a placeholder in the bpmn name that we are replacing. If i change dfspid to accountholdinginstitutionid then i have to reformat all bpmn process id

Comment (ankita10r) [reply src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java:182]: Upload bpmn logic works on this placeholder of dfspid so changing it here won't cause a major problem other than being in different from actual bpmn names

**Comment by ankita10r** [comment]
Replied in conversation. Dfspid here is merely a placeholder's carried on through bpmn name(upload bpmn logic uses it), so we are not confusing dfspid with tenant rather just replacing a placeholder

File: src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java (hunk 1 above)
Comment (fynmanoj) [reply src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java:182]: let's track this major change separately. please create a ticket in the backlog for the same

File: src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java [hunk 2]
```diff
@@ -59,6 +61,10 @@ public void setupWorkers() {
         workerSendPayeeFailureToChannel();
         workerInvokeAcknowledgementWorkflows();
         zeebeConsistencyWorker();
+<<<<<<< Updated upstream
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java:64]: please remove these conflict markers. Also make sure the branch is rebased properly

File: src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java [hunk 3]
```diff
@@ -251,4 +257,55 @@ private void zeebeConsistencyWorker() {
...
+                            .join();
+                })
+                .name("validate-transactional-data")
+                .maxJobsActive(workerMaxJobs)
+                .open();
+
+
+    }
+
+    private boolean verifyTxn(TransactionChannelRequestDTO channelRequestDTO) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java:293]: can we rename this method to validateTxn

File: src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java [hunk 4]
```diff
@@ -251,4 +257,55 @@ private void zeebeConsistencyWorker() {
...
+            if (payeeIdentifierType.equalsIgnoreCase("MSISDN") || payeeIdentifierType
+                    .equalsIgnoreCase("ACCOUNTID")) {
+                String payeeIdentity = channelRequestDTO.getPayee().getPartyIdInfo().getPartyIdentifier();
+                valid = payeeIdentity.matches("^[\\d*#+]+$");
+            }
+        }
+        else valid = !channelRequestDTO.getNote().contains("Duplicate transaction");
+        return valid;
+    }
+>>>>>>> Stashed changes
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java:310]: please remove these conflict markers. Also make sure the branch is rebased properly

**Review by fynmanoj — CHANGES_REQUESTED** [review]
please remove conflict markers. Also make sure the branch is rebased properly

File: src/main/java/org/mifos/connector/channel/zeebe/ZeebeVariables.java [hunk 5]
```diff
@@ -26,5 +27,9 @@ private ZeebeVariables() {
 
     public static final String NOTE = "note";
     public static final String TRANSFER_CREATE_FAILED = "transferCreateFailed";
+<<<<<<< Updated upstream
```
[the remark this answers is not in the document]
Comment (ankita10r) [reply src/main/java/org/mifos/connector/channel/zeebe/ZeebeVariables.java:30]: yes

File: src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java (hunk 2 above)
Comment (ankita10r) [reply src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java:64]: yes

File: src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java (hunk 3 above)
Comment (ankita10r) [reply src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java:293]: okay

File: src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java (hunk 4 above)
Comment (ankita10r) [reply src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java:310]: okay

File: src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java (hunk 1 above)
Comment (ankita10r) [reply src/main/java/org/mifos/connector/channel/camel/routes/GSMAChannelRouteBuilder.java:182]: okay

File: src/main/resources/application.yml [hunk 6]
```diff
@@ -1,12 +1,12 @@
 camel:
-  server-port: 5000
+  server-port: 5001
```
Comment (fynmanoj) [inline src/main/resources/application.yml:2]: why is this Port number changed?

**Pull request merged**